Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -92,63 +76,61 @@ public class ResumableUpload<TRequest>
+            Options = options;
+        }
 
-            this.Service = service;
-            this.Path = path;
-            this.HttpMethod = httpMethod;
-            this.ContentStream = contentStream;
-            this.ContentType = contentType;
+        /// <summary>
+        /// Creates a <see cref="ResumableUpload"/> instance for a resumable upload session which has already been initiated.
+        /// </summary>
+        /// <remarks>
+        /// See https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload#start-resumable for more information about initiating
+        /// resumable upload sessions and saving the session URI.
+        /// </remarks>
+        /// <param name="sessionUri">The session URI of the resumable upload session. Must not be null.</param>
+        /// <param name="contentStream">The data to be uploaded. Must not be null.</param>
+        /// <param name="options">The options for the upload operation. May be null.</param>
+        /// <returns>The instance which can be used to upload the specified content.</returns>
+        public static ResumableUpload CreateFromSessionUri(
+            Uri sessionUri,
+            Stream contentStream,
+            ResumableUploadOptions options = null)
+        {
+            sessionUri.ThrowIfNull(nameof(sessionUri));
+            return new ResumableUpload(contentStream, options) { UploadUri = sessionUri };
         }
 
         #endregion // Construction
 
         #region Properties
 
-        /// <summary>Gets or sets the service.</summary>
-        public IClientService Service { get; private set; }
 
         /// <summary>
-        /// Gets or sets the path of the method (combined with
-        /// <see cref="Google.Apis.Services.IClientService.BaseUri"/>) to produce 
-        /// absolute Uri. 
+        /// Gets the options used to control the resumable upload.
         /// </summary>
-        public string Path { get; private set; }
+        protected ResumableUploadOptions Options { get; }
 
-        /// <summary>Gets or sets the HTTP method of this upload (used to initialize the upload).</summary>
-        public string HttpMethod { get; private set; }
+        /// <summary>
+        /// Gets the HTTP client to use to make requests.
+        /// </summary>
+        internal ConfigurableHttpClient HttpClient { get; }
 
         /// <summary>Gets or sets the stream to upload.</summary>
-        public Stream ContentStream { get; private set; }
-
-        /// <summary>Gets or sets the stream's Content-Type.</summary>
-        public string ContentType { get; private set; }
+        public Stream ContentStream { get; }
 
         /// <summary>
         /// Gets or sets the length of the steam. Will be <see cref="UnknownSize" /> if the media content length is 
         /// unknown. 
         /// </summary>
-        private long StreamLength { get; set; }
+        internal long StreamLength { get; set; }

```

Comment:
I was trying to disturb/expose as little as possible with these changes, so I thought internal would be better than protected, but perhaps I was just being too cautious here. I'll change it to protected and update the triple-slash to not reference the private `UnknownSize` constant.